File: javascript/hamming/hamming.js
var Hamming = function() {};

Hamming.prototype.compute = function(a, b) {
  // Short circuit an exact match.
  if (a === b) { return 0; }
  if (a.length !== b.length) { throw "DNA strands must be of equal length."; }
  return [...a].reduce((pV, cV, i) => (cV == b[i]) ? pV : ++pV, 0);
};

module.exports = Hamming;

File: go/hamming/hamming.go
package hamming

import (
	"errors"
)

const testVersion = 5

func Distance(a, b string) (int, error) {
	if len(a) != len(b) {
		return 0, errors.New("inputs must be equal length")
	}

	distance := 0
	for i := 0; i < len(a); i++ {
		if a[i] != b[i] {
			distance++
		}
	}

	return distance, nil
}
